internal/client/conn_batch.go

```diff
@@ -0,0 +1,339 @@
+// Copyright 2025 TiKV Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"runtime"
+	"runtime/trace"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/tikv/client-go/v2/config"
+	"github.com/tikv/client-go/v2/internal/logutil"
+	"github.com/tikv/client-go/v2/metrics"
+	"github.com/tikv/client-go/v2/util"
+	"go.uber.org/zap"
+)
+
+type batchConn struct {
+	// An atomic flag indicates whether the batch is idle or not.
+	// 0 for busy, others for idle.
+	idle uint32
+
+	// batchCommandsCh used for batch commands.
+	batchCommandsCh        chan *batchCommandsEntry
+	batchCommandsConns     []*batchCommandsClient
```

Comment:
This looks not proper to be renamed `...Conns`. It's not a real connection, and the word *client* comes from the naming of goproto. It's the client of the stream established by calling gRPC stream API.